File: backend/src/ERPBPM.Domain/Entities/Employee.cs
using ERPBPM.Domain.Common;
using ERPBPM.Domain.Enums;

namespace ERPBPM.Domain.Entities;

public class Employee : BaseEntity
{
    public string EmployeeNumber { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public DateTime DateOfBirth { get; set; }
    public DateTime HireDate { get; set; }
    public Guid DepartmentId { get; set; }
    public Department Department { get; set; } = null!;
    public Guid PositionId { get; set; }
    public Position Position { get; set; } = null!;
    public decimal Salary { get; set; }
    public EmployeeStatus Status { get; set; } = EmployeeStatus.Active;
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? Country { get; set; }
    public string? EmergencyContact { get; set; }
    public string? EmergencyPhone { get; set; }
}

File: frontend-mobile/lib/features/hrm/domain/entities/employee.dart
import 'package:equatable/equatable.dart';

class Employee extends Equatable {
  final String id; // Changed from int to String for GUID
  final String firstName;
  final String lastName;
  final String email;
  final String phoneNumber;
  final DateTime hireDate;
  final String status;
  final String positionName;
  final String departmentName;
  final double? salary;

  const Employee({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.phoneNumber,
    required this.hireDate,
    required this.status,
    required this.positionName,
    required this.departmentName,
    this.salary,
  });

  String get fullName => '$firstName $lastName';

  @override
  List<Object?> get props => [
        id,
        firstName,
        lastName,
        email,
        phoneNumber,
        hireDate,
        status,
        positionName,
        departmentName,
        salary,
      ];
}
